agent/engine/containermetadata/metadata_manager_windows.go

```diff
@@ -0,0 +1,75 @@
+// +build windows
+// Copyright 2014-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package containermetadata
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/aws/amazon-ecs-agent/agent/api"
+)
+
+const (
+	mountPoint = `C:\ecs\metadata`
```

Comment:
Yes, %ProgramData% seems like a good place for this. The host uses this location for logs and there doesn't seem to be any reason to separate this by logged-in user.

There are other places in agent that use this folder, so consistency will probably lead to a better user experience. Unfortunately, I believe the other locations (such as the amazon-ecs-agent.ps1
 script) are hardcoded to the C:\ProgramData path.